src/sentry/integrations/analytics.py

```diff
@@ -1,127 +1,94 @@
+    id: int
+    organization_id: str
+    user_id: str | None = None
+    default_user_id: str
 
 
+@analytics.eventclass("integration.issue.status.synced")
 class IntegrationIssueStatusSyncedEvent(analytics.Event):
-    type = "integration.issue.status.synced"
-
-    attributes = (
-        analytics.Attribute("provider"),
-        analytics.Attribute("id"),
-        analytics.Attribute("organization_id"),
-    )
+    provider: str
+    id: int
+    organization_id: str
 
 
+@analytics.eventclass("integration.issue.assignee.synced")
 class IntegrationIssueAssigneeSyncedEvent(analytics.Event):
-    type = "integration.issue.assignee.synced"
-
-    attributes = (
-        analytics.Attribute("provider"),
-        analytics.Attribute("id"),
-        analytics.Attribute("organization_id"),
-    )
+    provider: str
+    id: int
+    organization_id: str
 
 
+@analytics.eventclass("integration.issue.comments.synced")
 class IntegrationIssueCommentsSyncedEvent(analytics.Event):
-    type = "integration.issue.comments.synced"
-
-    attributes = (
-        analytics.Attribute("provider"),
-        analytics.Attribute("id"),
-        analytics.Attribute("organization_id"),
-    )
+    provider: str
+    id: int
+    organization_id: str
 
 
+@analytics.eventclass("integration.repo.added")
 class IntegrationRepoAddedEvent(analytics.Event):
-    type = "integration.repo.added"
-
-    attributes = (
-        analytics.Attribute("provider"),
-        analytics.Attribute("id"),
-        analytics.Attribute("organization_id"),
-    )
+    provider: str
+    id: int
+    organization_id: str
 
 
+@analytics.eventclass("integration.resolve.commit")
 class IntegrationResolveCommitEvent(analytics.Event):
-    type = "integration.resolve.commit"
-
-    attributes = (
-        analytics.Attribute("provider"),
-        analytics.Attribute("id"),
-        analytics.Attribute("organization_id"),
-    )
+    provider: str | None
+    id: int
+    organization_id: str
 
 
+@analytics.eventclass("integration.resolve.pr")
 class IntegrationResolvePREvent(analytics.Event):
-    type = "integration.resolve.pr"
-
-    attributes = (
-        analytics.Attribute("provider"),
-        analytics.Attribute("id"),
-        analytics.Attribute("organization_id"),
-    )
+    provider: str | None
+    id: str
+    organization_id: str
 
 
+@analytics.eventclass("integration.stacktrace.linked")
 class IntegrationStacktraceLinkEvent(analytics.Event):
-    type = "integration.stacktrace.linked"
-
-    attributes = (
-        analytics.Attribute("provider"),
-        analytics.Attribute("config_id"),
-        analytics.Attribute("project_id"),
-        analytics.Attribute("organization_id"),
-        analytics.Attribute("filepath"),
-        analytics.Attribute("status"),
-        analytics.Attribute("link_fetch_iterations"),
-        analytics.Attribute("platform", required=False),
-    )
+    provider: str
+    config_id: str
+    project_id: int
+    organization_id: int
```

Comment:
is there a method to whether `id`'s are `string` or `int`? I saw `organization_id: str` just above this line.

My understanding is that especially for org/proj/user we should be treating them as strings in all or at least most places because some of them can overflow numeric types in certain languages (not python, just saying). Also we don't do arithmetic on id's, so string works as a great default.